src/coreclr/src/tools/crossgen2/ILCompiler.Reflection.ReadyToRun/ReadyToRunMethod.cs

```diff
@@ -148,7 +193,6 @@ public DebugInfo DebugInfo
             }
             CodeOffset = codeOffset;
             method.GcInfo = gcInfo;
```

Comment:
I have a strong suspicion that this logic (i.e. `method.GcInfo = gcInfo;`) was wrong (and it is still wrong, my change is a pure refactoring that does not fix this yet).

The `gcInfo` parameter is computed through `ParseRuntimeFunctionsForMethod`, it appears to me that it should be associated with the `RuntimeFunction`, not the `ReadyToRunMethod`.

According to `ParseRuntimeFunctionsForMethod`, all but the first `RuntimeFunction` associated with the method could have `GCInfo`, but all of them have `UnwindInfo`.

Reply:
I'm not sure about this - according to my recollection, JIT provides and Crossgen1/2 encodes just one GC info per method. Not sure if that means it covers all the secondary funclets or whether these somehow trigger a different GC behavior.

Reply:
Thanks for your feedback, it is very useful.

Basically, I was tripped by the name `RuntimeFunction`, thinking different instances of `RuntimeFunction` means separate functions. If this could trip me, this could trip anyone using the code, we need a better name/comment/documentation for it, TBD.

I made a small change to avoid showing duplicated GCInfo for each `RuntimeFunction`. This should reduce confusion and also trim down the output file size.